File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int,vector<int>,greater<int>> pq;
        for(auto i:nums){
            if(pq.size()!= k || i > pq.top()){
                if (pq.size() == k)
                    pq.pop();
                pq.push(i);
            }
        }
        return pq.top();
    }
};

File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        # make a max heap.
        for i in range(len(nums)):
            nums[i]*=-1
        heapq.heapify(nums)
        while(k-1):
            heapq.heappop(nums)
            k-=1
        return nums[0]*-1